Source/Charts/Renderers/ChartDataRendererBase.swift

```diff
@@ -55,7 +55,9 @@ open class DataRenderer: Renderer
     {
         guard let data = dataProvider?.data
             else { return false }
-        
-        return data.entryCount < Int(CGFloat(dataProvider?.maxVisibleCount ?? 0) * (viewPortHandler?.scaleX ?? 1.0))
+
+        let maxVisibleCount = CGFloat(dataProvider?.maxVisibleCount ?? 0)
+        let scale = self.viewPortHandler?.scaleX ?? 1.0
```

Comment:
The `self` change does account for most of the improvements. Chained operations do tack on some build time increase but it's probably not significant for this case. Happy to go ahead and update with that change.